core/src/main/java/com/aem/demo/core/models/NewsCardModel.java

```diff
@@ -39,7 +44,62 @@ public class NewsCardModel {
     @Default(values = DEFAULT_IMAGE)
     private String image;
 
+//    private int likes;
+
+    @Inject
+    private SlingHttpServletRequest servletRequest;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+
     public String getPath() {
-        return resource.getPath().replace("/jcr:content","");
+        return resource.getPath().replace("/jcr:content", "");
+    }
+
+    public String getFullPath() {
+        return resource.getPath();
+    }
+
+
+    public int  getLikes() {
+        int likes = 0;
+
+        String rootPath = resource.getPath();
+
+        resource = resourceResolver.getResource(rootPath);
+        ModifiableValueMap properties = resource != null ? resource.adaptTo(ModifiableValueMap.class) : null;
+
+        if (properties != null) {
+            likes = resource.adaptTo(ValueMap.class).get("likes", 0);
+            properties.put("likes", likes + 1);
```

Comment:
Here comes the major problem. 
Looks like the number of likes incremented every time the getter is called. As many times we update the page in the browser, as many times the getter is called, as many likes we get)
Must not work this way.
Violates the semantics of a `get...()` method.
Please refactor this core to create a dedicated `set...()` or else `update...()` method.

Reply:
Yes this is the problem thats why I did not wanted to push these changes yesterday cuz they were not completed but sorry I see that I messed it up a little bit)